lib/foreman/renderer/scope/macros/host_template.rb

```diff
@@ -52,6 +52,18 @@ def host_param!(param_name)
             host_param(param_name)
           end
 
+          apipie :method, 'Returns puppetserver configured for the host' do
+            returns String, desc: 'Returns puppetserver configured through ENC or puppetserver host param'
+          end
+          def host_puppetserver
+            check_host
+            host.try(:puppetserver) || host.try(:puppetmaster) || host_param('puppetserver')
+          end
+
+          def host_puppet_environment
```

Comment:
Just a question, why this method is not documented and doesn't have `check_host`?

Reply:
because stupid me... on it, thanks :+1:

Reply:
Done